### ISSUE TITLE
Assigning STAND_STRUCTURE when we don't have a full inventory

### ISSUE THREAD

**Issue by edwardsmarc** [edwardsmarc/CASFRI#291]
As far as I am aware, this will only be relevant in BC where we don't have access to the full inventory.

Note the VRI attribute **layer_id** records the layer order from 1 (tallest) to 4.

## BC10
We have access to: 
* VEG_COMP_LYR_L1_POLY - tallest layer, always layer_id = 1
* VEG_COMP_LYR_L2_POLY - second tallest layer, always layer_id = 2

#### LAYER
Currently we apply CASFRI LAYER values in order - for VRI layer 1, VRI layer 2, and a single VRI NFL layer - based on which layers have values. For example if VRI layer 1 is null, but we have VRI layer 2 and NFL information, we would assign the VRI layer 2 to CASFRI LAYER 1, and the NFL info to CASFRI LAYER 2.

#### STAND_STRUCTURE
We then assign STAND_STRUCTURE as 'S' if there is only 1 CASFRI layer, or 'M' if there are 2 or 3 CASFRI layers.
* **PROBLEM 1** - we don't know if there are also layers 3 and 4 present because we don't have access to that data. We could be incorrectly assigning 'S' when there could be more layers we don't know about. 

## BC08
We have access to:
* VEG_COMP_LYR_R1_POLY - rank 1 data, most important layer only, could be layer_id 1, 2, 3, 4.

#### LAYER
If we have canopy info we assign it to CASFRI LAYER 1, if we have NFL info we assign it to either CASFRI LAYER 1 or 2 depending on whether canopy info is present. 

#### STAND_STRUCTURE
We then assign STAND_STRUCTURE as 'S' if there is only 1 CASFRI layer, or 'M' if there are 2 CASFRI layers.
* **PROBLEM 1 also applies to BC08**

#### BC10
* Since we always have the two tallest layers, but no other layers, there is no way for us to know whether other layers exist. Our only option is to report 'S' in cases where we only have a single CASFRI layer.

#### BC08
* If we have canopy data and the layer_id is greater that 1, we know we are dealing with a multi-story polygon. We could apply STAND_STRUCTURE = 'M' in these cases, even though we only have a single CASFRI layer.
* If our canopy layer has a layer_id of 1 and we have no NFL info, or if we only have an NFL layer, we again have no way of knowing if other layers exist. We would have to report 'S'.  

#### Error code
* Another option is to report an error code in cases where we can't be certain of the correct STAND_STRUCTURE value.

**Comment by pedrogit** [comment]
Should we assign STAND_STRUCTURE based on the data we have or on the data we don't? Will it make sense to someone receiving CASFRI to see STAND_STRUCTURE = 'M' when there is only one layer?

**Comment by edwardsmarc** [comment]
I would agree STAND_STRUCTURE should be consistent only within CASFRI. Users should not need to refer to source inventories to understand CASFRI. In the case of BC we will lose some layer info with this approach, unless/until the entire inventory becomes available.

**Comment by edwardsmarc** [comment]
This question was raised by Sishir. Check this conclusion is acceptable.
